File: 07-eloop/isprime.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    long i, num;
    if (argc < 2)
    {
        return 1;
    }

    num = strtol(argv[1], NULL, 10);
    if (num < 2)
    {
        return 1;
    }
    for (i = 2; i < num; ++i)
    {
        if (num % i == 0)
        {
            return 1;
        }
    }
    return 0;
}

File: 07-eloop/isprime.mjs
#!/usr/bin/env node
import { isPrime } from "./cpu-intensive.mjs";

const num = parseInt(process.argv[2]);
if (isNaN(num)) {
    console.error(`${num} is not a number`);
    process.exit(2);
}

if (isPrime(num)) {
    console.log(`${num} is prime`);
    process.exit(0);
} else {
    console.log(`${num} is not prime`);
    process.exit(1);
}
